changes de yml to better reflect regions

chinito opened the pull request:
used car license plate abbreviations to reflect the regions where a
holiday is celebrated

Sorry, this doesn't fix the problem with missing holidays on de_by_a AKA de_by_aux, I'm still investigating.

chinito commented:
I'm going to update this PR if my PR on the gem is accepted

ptrimble commented:
Hi @chinito! Thanks for the submission. 

Can you help me understand the change here? Generally we are trying to use iso3166 keys (as mentioned [here](https://github.com/holidays/definitions/blob/master/SYNTAX.md#region)). If I understand correctly it seems like these specific regions that you are changing are not considered 'official' regions. I was going off of this list: https://en.wikipedia.org/wiki/ISO_3166-2:DE

I can respect this change! When we get into smaller, less 'official' regions we are left with subjective choices and I will always defer to native speakers. But could you help me understand why it is important to change? Once I understand I will merge and get it into the official gem. Thank you!

chinito commented:
Hi @ptrimble! Thanks for the review.

I spend some hours on the weekend to understand how this `_aux` is meant. At first my opinion was, that aux is an abbrevation for Augsburg (City I live in and my company is located), because some people write it sometimes with X, because they things Auxburg is more funky or whatever.
Than I saw `:de_sn_aux` and `:de_th_aux`. 
So this did make any sense to me, as in Sachsen and Thüringen there is no Augsburg.
After some Wikipedia searching I came to the solution, it's for Fronleichnam, which is there celebrated in catholic areas. So I thought maybe this is `_aux` should mean auxilliary or whatever.

So, to clear things up and hopefully make it understandable for a guy living in the USA. In Germany we have 16 States (called Länder), these are in somewhat similar to your states like Illionois, Texas, California. All of them have an own parliament and stuff. So it is a lot more than a region, but a state in the federal republic of Germany. Each of these States has all Germany Holidays besides the German wides for its own. Those often depend on religious traditions are sometimes only locally.
I can now speak for Bavaria and of course as I read about in Wikipedia for Sachsen and Thüringen.
Generally Wikipedia is very detailed about it: https://en.wikipedia.org/wiki/Public_holidays_in_Germany

So lets come back to the actual holiday problem I have (especially as CEO of my own Ruby on Rails Company with employees who have freetime ;).
We have all the holidays from bavaria `:de_by`plus only one day on the 8th of august, called Friedensfest (peace festival). This holiday goes back to a peace treaty between protestants and catholics about 400 years ago and is only with the border of the city of augsburg. If you go from my house to Stadtbergen (which is about 1,5 km) there will be no holiday on this day. So only round about 280_000 people have this specially holiday and the only have it, if there company is also located in Augsburg. So if you work in Munich, you don't have it, if a munich guy works in my town he has it.

Same is for that `:de_sn_aux`and 'de_th_aux` on Fronleichnam. This are only small parts of the states they belong to. Wikipedia says: 

TH: Public holiday only in few Sorbian communities.
SN :Public holiday only in the Catholic district of Eichsfeld.

So I renamed it to the locallities, because `aux`is not descripitve in any way. So I thought of something smaller on how to specify these small parts. So I cam up with the car plates. Obviously this is not really correct, as the towns around augsburg with no holiday on the 8th of august, share the same car plate. Also in Thüringen there are multiple car plates having Fronleichnam.

What do you think about the following?

`:de_by_aux` will be `:de_by_augsburg`
`:de_sn_aux`will be `:de_sn_eichsfeld`
`:de_th_aux`will be `:de_th_sorbian`


Also we have another special thing in bavaria (we are almost 13 mio). The 15th of August is Mariä Himmelfahrt. This is only celebrated in counties where there are more catholics than protestants. (2014 this changed in some parts to be more protestants, so people where not so happy).
We also should address this in some way. Maybe `:de_by_catholic`and `:de_by_protestant`would be solution?

Please let's think about this together, so we find a really good and appropriate solution. And unforntunatly "informal" is not the right attribute, as these are real free days in this parts.

ptrimble commented:
@chinito Wow! Thank you so much for this explanation. It makes a lot of sense to me. 👍 

I apologize for taking so long to respond...my first baby is due on December 16th so this is an especially crazy time! I have been slacking on my response times. 😄 

Firstly, I should confess something embarrassing: I had no idea that Augsburg was outside of Munich. I follow Bundesliga pretty religiously and know about FC Augsburg but....I had no idea it was so close to Bayern's turf! That's embarrassing on my part. 😃 

Now, onto the task at hand: your explanation makes _a lot_ of sense to me. I knew about `Länder` (although I didn't know the name, just the concept of 'states' under the German federal republic). I also understand that the rules for holidays can be complex depending on local rules. So, to start, I think that:

```
:de_by_aux will be :de_by_augsburg
:de_sn_auxwill be :de_sn_eichsfeld
:de_th_auxwill be :de_th_sorbian
```

Makes a lot of sense. When in doubt I like the idea of spelling out *exactly* what we mean, to cut down on confusion. So let's do that.

For the last part (the Bavaria part)...I'm honestly not sure. The idea of `:de_by_catholic` and `:de_by_protestant` feels incorrect to me, but only because we have generally stuck to geographic regions in our names and this is about how you identify as a person rather than where you live. 

So here is what I propose:

1. You change to `de_by_augsburg`, etc as we discussed. I can merge that and we can move forward and at least get that since it sounds like we are in agreement.
2. For the `catholic` and `protestant` ideas, can we continue that discussion? We can do it in a separate issue that I would request that you create, so I can learn a bit more about the counties in Germany. We have a similiar idea of 'counties' in states here in the USA (I like in 'Lake' county in 'Illinois', for example). So I lean towards us specifying the specific counties and then...just having to change them as necessary as time goes on. But I think a longer discussion would be useful and I would need to learn more.

So let's do that. Let's do number 1 now as part of this PR and then, if you don't mind, open an issue to talk about number 2. We can continue this discussion.

Thanks so much for taking the time to talk about this! I very much appreciate it and I also appreciate your patience and I try to understand. 😄

chinito commented:
@ptrimble So i updated this as suggested and also added a new issue for Mariä Himmelfahrt. Hope it explains well. Here is a map of Germany with all Holidays by color, the darker, the more holidays: https://upload.wikimedia.org/wikipedia/commons/a/a5/Feiertage_in_Deutschland.png

If you have time, please give me feedback on the additional test in the holiday gem.

Also I'm offering a wiki or readme guide on how to contribute with the definitions and not messing around with those as I did. 

congratiulations and good look to you and your wife on beiing parents soon!

ptrimble commented:
Thanks for your patience. Baby is not here yet but he's coming fast! 😛 

This looks good! I'll go ahead and merge it. Let's talk more in the coming weeks about how to handle the Catholic v Protestant breakdown.

ttwo32 commented:
@ptrimble Thank you for reviewing and merging this.:)
@chinito Thank you for your contribute.😄
